scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -0,0 +1,80 @@
+package com.twitter.scalding.beam_backend
+
+import com.twitter.scalding.Config
+import com.twitter.scalding.typed.TypedSource
+import org.apache.beam.sdk.Pipeline
+import org.apache.beam.sdk.transforms.DoFn.ProcessElement
+import org.apache.beam.sdk.transforms._
+import org.apache.beam.sdk.values.PCollection
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+sealed abstract class BeamOp[+A] {
+  import BeamOp.TransformBeamOp
+  def run(pipeline: Pipeline)(implicit ec: ExecutionContext): PCollection[_ <: A]
+
+  def map[C >: A, B](f: C => B): BeamOp[B] = {
+    this.parDo(new MapFn(f))
+  }
+  def parDo[C >: A, B](f: DoFn[C, B]): BeamOp[B] = {
+    this.applyPTransform(ParDo.of(f).asInstanceOf[PTransform[PCollection[C], PCollection[B]]])
+  }
+  def applyPTransform[C >: A, B](f: PTransform[PCollection[C], PCollection[B]]): BeamOp[B] = {
+    TransformBeamOp(this, f)
+  }
+  def filter(f: A => Boolean): BeamOp[A] = {
+    val fn = new ProcessFn(f.asInstanceOf[A => java.lang.Boolean])
+    this.applyPTransform(Filter.by[A, ProcessFunction[A, java.lang.Boolean]](fn))
```

Comment:
no need for `this` Do we need the cast and the type annotations here? seems like the cast alone should be good enough.

That said, I think doing `f.andThen(java.lang.Boolean.valueOf(_))` would safer than the cast which is relying on the scala implementation details which could potentially change in some versions (e.g. there is no specialization to boolean).